City-of-Helsinki/drupal-module-helfi-tpr
# UHF-4531: Address search changes

# Address search changes [UHF-4531](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4531)
Fixes some address search sorting issues and adds some requested changes.

## What was done
* Fixed sort order when showing entities and/or using pager.
* Shows the distance in kilometers. Shown minimum distance is 0,1 km.
* Added views area plugin for showing search information.

## How to install
* Make sure your instance is up and running on latest dev branch.
  * It might be easiest to test with the Kasko instance.
* Update the TPR module:
    * `composer require drupal/helfi_tpr:dev-UHF-4531-address-search-changes`
* Go to `/admin/modules` and enable "HELfi Address search".
* Edit some view that lists units, e.g. the new Daycare view at Kasko (see: [separate PR](https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/116)).
* Make sure the view has `Computed distance` field with default settings.
* Make sure the view has `Address search` filter with default settings.
* Check which template is used when showing the individual view results. For example, with the Kasko's Daycare view, the template file is `hdbt_subtheme/templates/module/helfi_tpr/tpr-unit.html.twig`.
  * Make sure the template has `{{ content.distance }}` for showing the distance.
* Clear caches: `drush cr`
* Make sure you have some page, which includes a paragraph for the unit search (e.g. Daycare search) and it shows some units.

## How to test
* Edit the view by adding a "Address search information" area to the header.
  * Add some informative text for successful and failed searches. 
* Go to the page that contains the view.
* Search with different addresses.
* Check that successful search shows the informative text for successful searches.
* Check that failed search (address not found) shows the informative text for failed searches.
* Check that the distance is shown as kilometers. Check that the smallest distance is 0,1 km.
* Check that results are sorted correctly: the nearest units are shown first.
* Check that the sorting works in different conditions:
  * When showing all results at once.
  * When using different pagers (e.g. pages and infinite scroll).
  * When cache is enabled.
